src/relay/backend/vm/lambda_lift.cc

```diff
@@ -111,22 +111,21 @@ class LambdaLifter : public transform::DeviceAwareExprMutator {
     auto free_type_vars = FreeTypeVars(func, module_);
 
     Array<Var> captured_vars;
-    std::vector<VirtualDevice> captured_var_virtual_devices;
     bool recursive = false;
     for (const auto& var : free_vars) {
       if (!letrec_.empty() && var == letrec_.back()) {
         recursive = true;
         continue;
       }
       captured_vars.push_back(var);
-      captured_var_virtual_devices.push_back(GetVirtualDevice(var));
```

Comment:
These vars may be let-bound as well as param-bound, so I don't think this is sound.

Reply:
Ah, so do I need to copy them and then substitute them in?